cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -107,6 +107,38 @@ csrfToken = hmac + "." + randomValue // Add the `randomValue` to the HMAC hash t
 response.setCookie("csrf_token=" + csrfToken + "; Secure") // Set Cookie without HttpOnly flag
 ```
 
+Below is an example in psuedo-code that demonstrates validation of the CSRF token once it is sent back from the client:
+
+```code
+// Get the CSRF token from the request
+csrfToken = request.getParameter("csrf_token") // From form field, cookie, or header
+
+// Split the token to get the randomValue
+const tokenParts = csrfToken.split(".");
+const hmacFromRequest = tokenParts[0];
+const randomValue = tokenParts[1];
+
+// Recreate the HMAC with the current session and the randomValue from the request
+secret = readEnvironmentVariable("CSRF_SECRET") // HMAC secret key
```

Comment:
Any need to link to some recommendations about the secret?

* randomly generated, with proper entropy
* not reused between environments
* not reused for another usage
* not reused for authenticating the data (unless some "usage" field is included which is not done currently)

Reply:
I think we have somwhere in cheatsheet about secrets generation so ideally we just link to that. We are avoiding duplicating content because than we will have instructions how to generate secrets in 3-4 places.

Reply:
I note in the explanation immediately above it has this, with a link to the key storage bit:

> Ideally, store this key as discussed in the [Cryptographic Storage page](https://cheatsheetseries.owasp.org/cheatsheets/Cryptographic_Storage_Cheat_Sheet.html#key-storage).